### ISSUE TITLE
Unexpectedly getting [Composer\Repository\InvalidRepositoryException].

### ISSUE THREAD

**Issue by rquadling** [composer/composer#4427]
Hi.

Attempting to update my projects dependencies and I'm getting an exception for a package that hasn't changed in a while.

Running with ```-vvv --profile```, I see ...

```
[5.1MB/4.25s] Executing command (C:/Users/RichardQ/AppData/Local/Composer/vcs/git-bitbucket.org-digitickets-tidypmp.git/): git rev-parse --git-dir
[5.1MB/4.36s] Executing command (CWD): git clone --mirror "git@bitbucket.org:digitickets/tidypmp.git" "C:/Users/RichardQ/AppData/Local/Composer/vcs/git-bitbucket.org-digitickets-tidypmp.git/"
[5.1MB/9.37s] Executing command (C:/Users/RichardQ/AppData/Local/Composer/vcs/git-bitbucket.org-digitickets-tidypmp.git/): git show-ref --tags
[5.1MB/9.42s] Executing command (C:/Users/RichardQ/AppData/Local/Composer/vcs/git-bitbucket.org-digitickets-tidypmp.git/): git branch --no-color --no-abbrev -v
[5.1MB/9.48s] Executing command (C:/Users/RichardQ/AppData/Local/Composer/vcs/git-bitbucket.org-digitickets-tidypmp.git/): git branch --no-color
[5.2MB/9.53s] Executing command (C:/Users/RichardQ/AppData/Local/Composer/vcs/git-bitbucket.org-digitickets-tidypmp.git/): git show "master":composer.json



  [Composer\Repository\InvalidRepositoryException]
  No valid composer.json was found in any branch or tag of git@bitbucket.org:digitickets/tidypmp.git, could not load a package from it.
```

Looking at the last git command being executed, I get ...
```
[ 2015/09/16 11:27:26 ] [ C:\Users\RichardQ\AppData\Local\Composer\vcs\git-bitbucket.org-digitickets-tidypmp.git ] >git show "master":composer.json
{
  "name": "digitickets/tidypmp",
  "type": "library",
  "description": "TidyPMP provides a mechanism to reformat the PHP Maker .pmp files allowing multiple developers better control when handling merge conflicts.",
  "keywords": [
    "tidypmp",
    "PHPMaker",
    "PHP Maker"
  ],
  "homepage": "https://bitbucket.org/digitickets/tidypmp",
  "license": "MIT",
  "authors": [
    {
      "name": "Richard Quadling",
      "email": "RQuadling@GMail.com",
      "role": "Lead Developer"
    }
  ],
  "require": {
    "rquadling/lalit": "~1.0",
    "rquadling/console": "~1.0",
    "symfony/console": "~2.6"
  },
  "autoload": {
    "psr-4": {
      "TidyPMP\\": "src/TidyPMP"
    }
  },
  "bin": [
    "bin/tidypmp"
  ]
}
```

Which is unchanged since 2015/04/29 and we've had many updates since then. Essentially, the composer.json file for this package seems to be fine and has been in play for many months with many subsequent updates.

Tried rolling composer backwards a few times. No avail.

Running as an admin.

Downloaded composer source and ran the tests. Getting some failures ...

```
PHPUnit 4.8.6 by Sebastian Bergmann and contributors.

............................................................   60 / 1040 (  5%)
............................................................  120 / 1040 ( 11%)
............................................................  180 / 1040 ( 17%)
............................................................  240 / 1040 ( 23%)
............................................................  300 / 1040 ( 28%)
............................................................  360 / 1040 ( 34%)
............................................................  420 / 1040 ( 40%)
...............................ES.E.........................  480 / 1040 ( 46%)
............................................................  540 / 1040 ( 51%)
............................................................  600 / 1040 ( 57%)
............................................................  660 / 1040 ( 63%)
............................................................  720 / 1040 ( 69%)
............................................................  780 / 1040 ( 75%)
............................................................  840 / 1040 ( 80%)
............................................................  900 / 1040 ( 86%)
............................................................  960 / 1040 ( 92%)
............................................................ 1020 / 1040 ( 98%)
....................

Time: 20.23 seconds, Memory: 22.75Mb

There were 2 errors:

1) Composer\Test\Package\Archiver\ArchivableFilesFinderTest::testGitExcludes
ErrorException: unlink(C:/Users/RichardQ/AppData/Local/Temp/composer_archiver_test115422051855f94d674df1f6.14462801/archive.zip): No such file or directory

D:\Sources\PHP\Composer\tests\Composer\Test\Package\Archiver\ArchivableFilesFinderTest.php:273
D:\Sources\PHP\Composer\tests\Composer\Test\Package\Archiver\ArchivableFilesFinderTest.php:200

2) Composer\Test\Package\Archiver\ArchiveManagerTest::testArchiveTar
RuntimeException: Could not commit: fatal: pathspec 'composer.json' did not match any files


D:\Sources\PHP\Composer\tests\Composer\Test\Package\Archiver\ArchiveManagerTest.php:102
D:\Sources\PHP\Composer\tests\Composer\Test\Package\Archiver\ArchiveManagerTest.php:43

FAILURES!
Tests: 1040, Assertions: 1836, Errors: 2, Skipped: 1.
```
Any ideas?

**Comment by rquadling** [comment]
Seems something has changed with regard to the execution of this command:
```
cmd /V:ON /E:ON /C "(git show "master":composer.json) 1>"C:\Users\RichardQ\AppData\Local\Temp\sf_A722.tmp" 2>"C:\Users\RichardQ\AppData\Local\Temp\sf_A723.tmp""
```

results in the error file containing ...
```
fatal: Invalid object name 'master'.
```

But, running ```git show "master":composer.json``` results in the right output.

**Comment by rquadling** [comment]
Very stuck.

**Comment by rquadling** [comment]
Aha!!!

```cmd /V:ON /E:ON /D /C "(git show "master":composer.json)```

works.

Cause is that the admin CMD has a CD in the registry to change directory. So when a new shell is loaded, the directory is changed away from where it needs to be.

Going to add a PR to Symfony\Process.

**Comment by rquadling** [comment]
https://github.com/symfony/Process/pull/9

**Comment by alcohol** [comment]
You should move your PR to `symfony/symfony`, but glad you resolved the issue :-).

**Comment by rquadling** [comment]
Yep. I did. https://github.com/symfony/symfony/pull/15817

**Comment by alcohol** [comment]
Closing this. Feel free to re-open though if the need arises.

**Comment by Seldaek** [comment]
Oh wow I can't believe it took so long for someone to notice this /D flag. Great that it's fixed though, thanks.

**Comment by rquadling** [comment]
Ha! Well, I don't think many people have autorunning registry entries that upset the status enough to worry about. Mine was changing directory so upset Symfony/Process. Quite pleased with myself for ~~causing~~ finding it.

**Comment by Seldaek** [comment]
Well a lot of people actually have outdated/invalid autorun entries that cause failures, we even have https://getcomposer.org/doc/articles/troubleshooting.md#-the-system-cannot-find-the-path-specified-windows- because of it ;)

**Comment by rquadling** [comment]
Please can you update the Symfony/Process you are using so this issue can be closed properly?

I've updated my composer and this issue is back. Having decompiled the composer.phar, I can see that the Process.php does NOT have the ``/D`` fix.

``Composer version 1.0-dev (a9f74803275a64273b73d7398a907d1a0e32c9a9) 2015-10-27 10:31:31``

The lock file for composer does suggest this should up fixed.

But ...

``C:\ > which composer`` shows

``
C:\ProgramData\ComposerSetup\bin\composer
C:\ProgramData\ComposerSetup\bin\composer.bat
```

``
cd /D C:\ProgramData\ComposerSetup\bin
md phar
php -r "(new Phar('./composer.phar'))->extractTo('./phar');"
grep cmd phar\vendor\symfony\process\Symfony\Component\Process\Process.php
```

outputs ...

$commandline = 'cmd /V:ON /E:ON /C "('.$commandline.')';

Sad face! :-(

**Comment by Seldaek** [comment]
Fixed, we were stuck on symfony 2.6 because they bumped their requirement to 5.3.9 in 2.7, but I fixed that.

**Comment by rquadling** [comment]
Thank you!!! Happiness! :smiley:
